services/src/main/java/org/keycloak/broker/saml/mappers/UserAttributeMapper.java

```diff
@@ -151,6 +152,12 @@ private void setIfNotEmpty(Consumer<String> consumer, List<String> values) {
         }
     }
 
+    private void setIfNotEmptyAndDifferent(Consumer<String> consumer, Supplier<String> currentValueSupplier, List<String> values) {
+        if (values != null && !values.isEmpty() && !values.get(0).equals(currentValueSupplier.get())) {
+            consumer.accept(values.get(0));
```

Comment:
I'm slightly worried about using the `0` index here. Are we sure, this is always going to be `0`?

Reply:
.get(0) - that is how setIfNotEmpty() was already implemented.
We also have checks - values != null && !values.isEmpty() that guarantee that collection is not empty.
Secondly, "get(0)" element is guaranteed to be non-null because "values" come from findAttributeValuesInContext() method which has ".filter(Objects::nonNull)"

Reply:
Fair enough, let me approve.